sdk/enter.go

```diff
@@ -80,6 +82,73 @@ func simpleChrootHelper(args []string) error {
 		return fmt.Errorf("Mounting %q failed: %v", newRepoRoot, err)
 	}
 
+	// mount the /run directory and setup the permissions
+	if err := syscall.Mount(
+		"tmpfs", "/run", "tmpfs", syscall.MS_NOSUID|syscall.MS_NODEV, "mode=755"); err != nil {
+		return fmt.Errorf("Mounting %q failed: %v", "/run", err)
+	}
+
+	userInfo, err := user.Lookup(username)
+	if err != nil {
+		return err
+	}
+
+	rundir := filepath.Join(chroot, "/run/user", userInfo.Uid)
+
+	err = os.MkdirAll(rundir, 0755)
+	if err != nil {
+		return err
+	}
+
+	uid, err := strconv.Atoi(userInfo.Uid)
+	if err != nil {
+		return err
+	}
+
+	gid, err := strconv.Atoi(userInfo.Gid)
+	if err != nil {
+		return err
+	}
+
+	err = os.Chown(rundir, uid, gid)
+	if err != nil {
+		return err
+	}
+
+	// mount the directory containing the ssh-agent socket in order
+	// to support a private repos during repo sync
+	sshAuthSock := os.Getenv("SSH_AUTH_SOCK")
+	if sshAuthSock != "" {
+		sshSourceDir, sshSocketFile := filepath.Split(sshAuthSock)
+		if _, err := os.Stat(sshSourceDir); err == nil {
+
+			// remove old ssh- directories
+			olddirs,_ := filepath.Glob(filepath.Join(rundir, "ssh-*"))
+			rm := exec.Command("rm", append([]string{"-rf"}, olddirs...)...)
+			if err := rm.Run(); err != nil {
+				return err
+			}
+
+			sshTargetdir, err := ioutil.TempDir(rundir, "ssh-")
+			if err != nil {
+				return err
+			}
+
+			err = os.Chown(sshTargetdir, uid, gid)
```

Comment:
Since this is a mount target the ownership of the underlying directory doesn't really matter, it will be hidden by the mount point.